### ISSUE TITLE
Accept right-aligned numerics

### ISSUE THREAD

**Issue by forthrin** [rubocop/rubocop#15286]
### Before submitting

- [x] I have searched the existing issues and this hasn't been reported yet.
- [x] I am running the latest stable version of RuboCop.

### Issue type

False positive

### Cop name

Layout/SpaceAroundOperators

### Steps to reproduce the problem

```ruby
if    a <  1 then 2
elsif a <  3 then a <▒ 4 + 20
elsif a < 10 then a < 10 +▒ 5
end
```

### Expected behavior

Accept right-aligned numbers.

### Actual behavior

Is the parser single-pass, no-look-ahead? Then this might be partially impossible.

### RuboCop configuration

```yaml

```

### RuboCop version

```shell
System Version: macOS 26.5.1 (25F80)
Model Identifier: MacBookAir10,1
ruby 4.0.5 (2026-05-20 revision 64336ffd0e) +PRISM [arm64-darwin25]
rubocop     2026-06-13 377d9ed06
```

**Comment by forthrin** [comment]
## AI assessment

This is a reasonable formatting exception, but it should be handled as alignment—not as a general relaxation of operator spacing.

Recommended strategy:

```yaml
Layout/SpaceAroundOperators:
  AllowForAlignment: true
```

When enabled, permit extra spaces only when they align corresponding expressions, such as numeric columns:

```ruby
if    a <  1 then 2
elsif a <  3 then a < 4 + 20
elsif a < 10 then a < 10 + 5
end
```

Still reject arbitrary extra spacing:

```ruby
result = a  +  b
```

The cop should determine alignment using neighboring lines and source columns, not merely accept multiple spaces everywhere. This keeps ordinary operator formatting strict while supporting intentional visual alignment.

This is a worthwhile, low-cost feature if implemented as an opt-in configuration option. It should not be inferred broadly from numeric literals alone.
